plugins/catalog-backend/src/stitching/progressTracker.ts

```diff
@@ -54,7 +54,9 @@ export function progressTracker(knex: Knex, logger: LoggerService) {
   stitchingQueueCount.addCallback(async result => {
     const total = await knex<DbRefreshStateRow>('refresh_state')
       .count({ count: '*' })
-      .whereNotNull('next_stitch_at');
+      .whereNotNull('next_stitch_at')
+      .whereNotNull('next_stitch_ticket')
```

Comment:
I think maybe leave out this condition.

If anything it should be `whereNull`, since that could potentially mean "hasn't been picked up yet for processing and is therefore still in the queue". The ticket gets set as a marker for someone starting to work on it.

But I'm not 100% sure that the semantics of having a ticket check will be the right one 🤔 because what happens when there's long running things that keep timing out? In those cases the tickets just hang around as non-null constantly.

I think for now, just remove this line, which will make it measure things that are either late OR busy being handled. As long as runtimes per task are fairly short, this shouldn't be an issue. The person monitoring this is mainly interested in seeing this graph being consistently high or rising, and alert on that. It's OK that it's "jagged" and stays low-ish.

Reply:
Oh right! I'll adjust accordingly. I didn't know exact semantics about `next_stitch_ticket` and just copied whatever the default stitcher has. Thanks for the clarification!